src/main/java/seedu/toluist/commons/util/DateTimeFormatterUtil.java

```diff
@@ -39,6 +55,104 @@ public static String formatEventRange(LocalDateTime from, LocalDateTime to) {
         }
     }
 
+    /**
+     * Format recurring task with deadline
+     */
+    public static String formatRecurringTaskDeadline(LocalDateTime deadline, LocalDateTime recurringEndDateTime,
+            RecurringFrequency recurringFrequency) {
+        String formattedResult = "";
+        switch (recurringFrequency) {
+        case DAILY:
+            formattedResult += DAILY + RECURRING_DATE_SEPARATOR + formatTime(deadline);
+            break;
+        case WEEKLY:
+            formattedResult += WEEKLY + RECURRING_DATE_SEPARATOR + formatDayOfWeek(deadline);
+            break;
+        case MONTHLY:
+            formattedResult += MONTHLY + RECURRING_DATE_SEPARATOR + formatDayOfMonth(deadline);
+            break;
+        case YEARLY:
+            formattedResult += YEARLY + RECURRING_DATE_SEPARATOR + formatDateOfYear(deadline);
+            break;
+        default:
+            // won't happen
+        }
+        if (recurringEndDateTime != null) {
+            formattedResult += RECURRING_DATE_SEPARATOR + UNTIL
+                    + RECURRING_DATE_SEPARATOR + formatDate(recurringEndDateTime);
+        }
+        return formattedResult;
+    }
+
+    /**
+     * Format recurring event
+     */
+    public static String formatRecurringEvent(LocalDateTime from, LocalDateTime to, LocalDateTime recurringEndDateTime,
+            RecurringFrequency recurringFrequency) {
+        String formattedResult = "";
+        switch (recurringFrequency) {
+        case DAILY:
+            formattedResult += DAILY;
```

Comment:
I just realized these 2 lines case DAILY: formattedResult += DAILY refers to different things. First DAILY refers to the enum while second DAILY refers to the string constant on top of this file. Is there a better way to handle this lol